apps/passport-client/new-components/screens/Home/NewHomeScreen.tsx

```diff
@@ -0,0 +1,456 @@
+      <InnerContainer>
+        <Typography fontWeight={800} color="var(--text-tertiary)">
+          YOU HAVE NO EVENT PASSES
+        </Typography>
+        <Typography color="var(--text-tertiary)">
+          Make sure you are logged in with the correct email address.
+        </Typography>
+      </InnerContainer>
+    </EmptyCardContainer>
+  );
+};
+
+export const NewHomeScreen = (): ReactElement => {
+  useSyncE2EEStorage();
+  const tickets = useTickets();
+  const [currentPos, setCurrentPos] = useState(0);
+  const [eventCardWidth, setEventCardWidth] = useState(0);
+  const [ticketCardWidth, setTicketCardWidth] = useState(0);
+  const [ticketsColumnHeight, setTicketsColumnHeight] = useState(0);
+  const dispatch = useDispatch();
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const pcdCardScrollRef = useRef<HTMLDivElement>(null);
+  const ticketsRef = useRef<Map<string, HTMLDivElement[]>>(new Map());
+  const windowWidth = useWindowWidth();
+  const self = useSelf();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!self) {
+      navigate("/new/login", { replace: true });
+    }
+  });
+
+  useLayoutEffect(() => {
+    if (scrollRef.current) {
+      setEventCardWidth(
+        calculateElWidth(
+          scrollRef.current.scrollWidth,
+          EVENT_GAP,
+          tickets.length
+        )
+      );
+    }
+    if (pcdCardScrollRef.current) {
+      setTicketCardWidth(
+        calculateElWidth(
+          pcdCardScrollRef.current.scrollWidth,
+          TICKETS_HORIZONTAL_GAP,
+          tickets.length
+        )
+      );
+    }
+  }, [setEventCardWidth, setTicketCardWidth, tickets.length]);
+
+  useEffect(() => {
+    if (tickets[currentPos]) {
+      const refs = ticketsRef.current.get(tickets[currentPos][0]);
+      if (!refs) {
+        setTicketsColumnHeight(0);
+        return;
+      }
+      setTicketsColumnHeight(calculateTicketsColumnHeight(refs));
+    }
+  }, [currentPos, tickets]);
+
+  const renderedTickets = useMemo(() => {
+    ticketsRef.current = new Map();
+    return tickets.map(([eventName, eventTicketPack]) => {
+      return (
+        <TicketsContainer
+          key={eventTicketPack.map((pack) => pack.eventTicket.id).join("-")}
+        >
+          {eventTicketPack.map((pack) => {
+            return (
+              <CardBody
+                key={pack.eventId}
+                addOns={
+                  pack.addOns.length > 0
+                    ? {
+                        text: `View ${pack.addOns.length} add-on items`,
+                        onClick(): void {
+                          dispatch({
+                            type: "set-bottom-modal",
+                            modal: {
+                              addOns: pack.addOns,
+                              modalType: "ticket-add-ons"
+                            }
+                          });
+                        }
+                      }
+                    : undefined
+                }
```

Comment:
are you adding 'addOns' in the `initTestData` function? Would be helpful for local development to do that so that I can see it work locally.